commcare_connect/form_receiver/processor.py

```
@@ -250,18 +252,52 @@ def process_assessments(user, xform: XForm, app: CommCareApp, opportunity: Oppor
+            work_area = WorkArea.objects.select_for_update().get(case_id=work_area_case_id, opportunity=opportunity)
+        except WorkArea.DoesNotExist:
+            raise ProcessingError("Work area not found")
+
+        if not work_area.work_area_group or work_area.work_area_group.opportunity_access_id != access.id:
+            raise ProcessingError("User is not assigned to this work area")
+
+        requested_status = block.get("status", "").upper()
+        try:
+            new_status = WorkAreaStatus(requested_status)
+        except ValueError:
+            raise ProcessingError(f"Invalid work area status: {requested_status}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)